pkg/codegen/dotnet/gen.go

```diff
@@ -2238,12 +2239,31 @@ func genProjectFile(pkg *schema.Package,
 		restoreSources = strings.Join(folders.ToSlice(), ";")
 	}
 
-	// Add the Pulumi package reference
+	// Add local package references
 	for _, path := range localDependencies {
-		filename := filepath.Base(path)
-		pkg, rest, _ := strings.Cut(filename, ".")
-		version, _ := strings.CutSuffix(rest, ".nupkg")
-		packageReferences[pkg] = version
+		nugetFile := ""
+		if strings.HasSuffix(path, ".nupkg") {
```

Comment:
This should always point directly to the nuget. We don't need to support it pointing to a directory.

Reply:
Ahh does that mean that `Pack` should return `ArtifactPath` being the path to the nuget itself? Because I was returning the containing directory

Reply:
Yup I made that comment on your other PR as well :)

Reply:
Fixed it ✅